core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -417,6 +423,38 @@ public void reconfigure(ServerConfigurator serverConfigurator) {
         pipelineConfigurator.updateConfig(config);
     }
 
+    /**
+     * Registers a JVM shutdown hook that closes this {@link Server} when the current JVM terminates.
+     */
+    public CompletableFuture<Void> closeOnShutdown() {
+        return closeOnShutdown(null);
+    }
+
+    /**
+     * Registers a JVM shutdown hook that closes this {@link Server} when the current JVM terminates.
+     *
+     * @param whenClosing {@link Runnable} will be run before closing this {@link Server}
+     */
+    public CompletableFuture<Void> closeOnShutdown(@Nullable Runnable whenClosing) {
+        final CompletableFuture<Void> future = new CompletableFuture<>();
+        Runtime.getRuntime().addShutdownHook(THREAD_FACTORY.newThread(() -> {
+            if (whenClosing != null) {
+                whenClosing.run();
```

Comment:
Should we wrap `whenClosing` with `try..catch` block and leave a warning for an exception?